doc/source/rllib/rllib-offline.rst

```diff
@@ -8,217 +8,662 @@ Working With Offline Data
+It should take you around 98 seconds (456 iterations) to achieve the same episode return mean as the PPO agent. While this may not seem 
+impressive compared to the PPO training time, it is important to note that ``CartPole-v1`` is a very simple environment to learn. In more 
+complex environments, which require more sophisticated agents and significantly longer training times, pretraining via behavior cloning 
+can be highly beneficial. Combining behavior cloning with subsequent fine-tuning using a reinforcement learning algorithm can substantially 
+reduce training time, resource consumption, and associated costs.
 
-.. note:: For a contextual bandit dataset, the ``dones`` key should always be set to ``True``. In this case, FQE reduces to fitting a reward model to the data.
+Using external expert experiences
+---------------------------------
+Your expert data is often already available, either recorded from an operational system or directly provided by human experts. Typically, 
+this data is stored in a tabular (columnar) format. RLlib's new Offline RL API simplifies the use of such data by allowing direct ingestion 
+via a specified schema that organizes the expert data. The API's default schema for reading data is provided in `offline_prelearner.SCHEMA 
+<http://github.com/ray-project/ray/blob/master/rllib/offline/offline_prelearner.py#33>`__.
 
-RLlib's OPE estimators output six metrics:
+Lets consider a simple example in which your expert data is stored with the schema: ``(o_t, a_t, r_t, o_tp1, d_t, i_t, logprobs_t)``. In this case 
+you provide this schema as follows:
 
-- ``v_behavior``: The discounted sum over rewards in the offline episode, averaged over episodes in the batch.
-- ``v_behavior_std``: The standard deviation corresponding to v_behavior.
-- ``v_target``: The OPE's estimated discounted return for the target policy, averaged over episodes in the batch.
-- ``v_target_std``: The standard deviation corresponding to v_target.
-- ``v_gain``: ``v_target / max(v_behavior, 1e-8)``. ``v_gain > 1.0`` indicates that the policy is better than the policy that generated the behavior data. In case, ``v_behavior <= 0``, ``v_delta`` should be used instead for comparison.
-- ``v_delta``: The difference between v_target and v_behavior.
+.. code-block:: python
 
-As an example, we generate an evaluation dataset for off-policy estimation:
+    from ray.rllib.algorithms.bc import BCConfig
+    from ray.rllib.core.columns import Columns
 
-.. code-block:: bash
+    config = (
+        BCConfig()
+        ...
+        .offline_data(
+            input_=[<input_path>],
+            # Provide the schema of your data.
+            input_read_schema={
+                Columns.OBS: "o_t",
+                Columns.ACTIONS: "a_t",
+                Columns.REWARDS: "r_t",
+                Columns.NEXT_OBS: "o_tp1",
+                Columns.INFOS: "i_t",
+                "done": "d_t",
```

Comment:
Use `Columns.DONES`?

Or should we even use `TERMINATEDS` here? (not sure what effects using "dones" would have on the algo loss logic).

Reply:
We don't have `Columns.DONES` :) I can create them.

Reply:
Nope, that's fine! Let's leave it as-is with this comment here.